### PR TITLE
rdp-server: add timer management within the eventloop

### PR DESCRIPTION
**Pull request by hardening**
Implement timers natively in the ogon eventloop and change existing code to using them.

### CODE REVIEW

**Comment by hardening** [comment]
Super interesting PR !

**Comment by hardening** [comment]
@nfedera @bmiklautz anyone to have a look at that interesting PR that simplifies the code with timers a lot (and have unitary tests !) ?

**Comment by hardening** [comment]
This is really good stuff, don't hesitate to review and merge !

**Comment by bmiklautz** [comment]
@hardening sorry for the long delay but finally on it

As a start the new Test fails on my machine with a counter of `0` (at return 10 in the test)

**Comment by bmiklautz** [comment]
@hardening what FreeRDP version is required for it?

File: rdp-server/test/TestOgonTimer.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+int TestOgonTimer(int argc, char* argv[])
+{
+	OGON_UNUSED(argc);
+	OGON_UNUSED(argv);
+	ogon_event_loop *loop;
+	ogon_event_source *evsrc;
+	int counter = 0;
+	DWORD endDate;
```
Comment (bmiklautz) [inline rdp-server/test/TestOgonTimer.c]: this should be a ULONGLONG (or UINT64)

**Review by bmiklautz — CHANGES_REQUESTED** [review]

**Comment by bmiklautz** [comment]
@hardening thank you for the nice addition! - Just tested it with base connections and shadowing. With the one remark is fixed it's good to go for me!

**Pull request merged**